src/nodes.coffee

```diff
@@ -1200,6 +1186,35 @@ exports.Value = class Value extends Base
     else
       @error 'tried to assign to unassignable value'
 
+  ast: ->
+    # If the `Value` has no properties, the AST node is just whatever this
+    # node’s `base` is.
+    return @base.ast() unless @hasProperties()
+    super()
+
+  astType: -> 'MemberExpression'
```

Comment:
Thanks, I like this new version. My only edit was to add a comment so that it’s clear why we’re overriding _both_ `ast` and the `ast*` methods. Otherwise this looks great.